tools/sim/bridge.py

```diff
@@ -231,266 +234,290 @@ def can_function_runner(vs: VehicleState, exit_event: threading.Event):
-      steer_op = sm['carControl'].actuators.steeringAngleDeg
-
-      throttle_out = throttle_op
-      steer_out = steer_op
-      brake_out = brake_op
-
-      steer_out = steer_rate_limit(old_steer, steer_out)
-      old_steer = steer_out
-
-    else:
-      if throttle_out == 0 and old_throttle > 0:
-        if throttle_ease_out_counter > 0:
-          throttle_out = old_throttle
-          throttle_ease_out_counter += -1
-        else:
-          throttle_ease_out_counter = REPEAT_COUNTER
-          old_throttle = 0
-
-      if brake_out == 0 and old_brake > 0:
-        if brake_ease_out_counter > 0:
-          brake_out = old_brake
-          brake_ease_out_counter += -1
-        else:
-          brake_ease_out_counter = REPEAT_COUNTER
-          old_brake = 0
-
-      if steer_out == 0 and old_steer != 0:
-        if steer_ease_out_counter > 0:
-          steer_out = old_steer
-          steer_ease_out_counter += -1
-        else:
-          steer_ease_out_counter = REPEAT_COUNTER
-          old_steer = 0
-
-    # --------------Step 2-------------------------------
-    steer_carla = steer_out / (max_steer_angle * STEER_RATIO * -1)
-
-    steer_carla = np.clip(steer_carla, -1, 1)
-    steer_out = steer_carla * (max_steer_angle * STEER_RATIO * -1)
-    old_steer = steer_carla * (max_steer_angle * STEER_RATIO * -1)
-
-    vc.throttle = throttle_out / 0.6
-    vc.steer = steer_carla
-    vc.brake = brake_out
-    vehicle.apply_control(vc)
-
-    # --------------Step 3-------------------------------
-    vel = vehicle.get_velocity()
-    speed = math.sqrt(vel.x**2 + vel.y**2 + vel.z**2)  # in m/s
-    vehicle_state.speed = speed
-    vehicle_state.vel = vel
-    vehicle_state.angle = steer_out
-    vehicle_state.cruise_button = cruise_button
-    vehicle_state.is_engaged = is_openpilot_engaged
-
-    if rk.frame % PRINT_DECIMATION == 0:
-      print("frame: ", "engaged:", is_openpilot_engaged, "; throttle: ", round(vc.throttle, 3), "; steer(c/deg): ", round(vc.steer, 3), round(steer_out, 3), "; brake: ", round(vc.brake, 3))
-
-    if rk.frame % 5 == 0:
-      world.tick()
-
-    rk.keep_time()
-
-  # Clean up resources in the opposite order they were created.
-  exit_event.set()
-  for t in reversed(threads):
-    t.join()
-  gps.destroy()
-  imu.destroy()
-  for c in cameras:
-    c.destroy()
-  vehicle.destroy()
-
-
-def bridge_keep_alive(q: Any):
-  while 1:
-    try:
-      bridge(q)
-      break
-    except RuntimeError as e:
-      print("Restarting bridge. Error:", e)
+      if rk.frame % 5 == 0:
+        world.tick()
+      rk.keep_time()
+    # Clean up resources in the opposite order they were created.
+    exit_event.set()
+    for t in reversed(threads):
+      t.join()
+    gps.destroy()
+    imu.destroy()
+    for c in cameras:
+      c.destroy()
+    vehicle.destroy()
+
+
+def main(add_args=None, keep_alive=True):
+  unblock_stdout()  # Fix error when publishing too many lag message
+  _args = parse_args(add_args)
```

Comment:
why is this prefixed with an underscore? usually, this is for private variables in python

Reply:
Changed it. Was still there after moving it from the main if statement